app/compliance.go

```diff
@@ -16,19 +16,25 @@ func (a *App) GetComplianceReports(page, perPage int) (model.Compliances, *model
 		return nil, model.NewAppError("GetComplianceReports", "ent.compliance.licence_disable.app_error", nil, "", http.StatusNotImplemented)
 	}
 
-	return a.Srv().Store.Compliance().GetAll(page*perPage, perPage)
+	compliance, err := a.Srv().Store.Compliance().GetAll(page*perPage, perPage)
+
+	if err != nil {
+		return nil, model.NewAppError("GetComplianceReports", "app.compliance.save_compliance.app_error", nil, err.Error(), http.StatusNotImplemented)
+	}
+	return compliance, nil
 }
 
 func (a *App) SaveComplianceReport(job *model.Compliance) (*model.Compliance, *model.AppError) {
 	if license := a.Srv().License(); !*a.Config().ComplianceSettings.Enable || license == nil || !*license.Features.Compliance || a.Compliance() == nil {
-		return nil, model.NewAppError("saveComplianceReport", "ent.compliance.licence_disable.app_error", nil, "", http.StatusNotImplemented)
+		return nil, model.NewAppError("saveComplianceReport", "app.compliance.licence_disable.app_error", nil, "", http.StatusNotImplemented)
 	}
 
 	job.Type = model.COMPLIANCE_TYPE_ADHOC
 
 	job, err := a.Srv().Store.Compliance().Save(job)
+
 	if err != nil {
```

Comment:
Save can return different kind of errors, actually it can return a `model.AppError` from the `IsValid` check of the compliance, or a regular error, you should handle the regular errors like you have done here, but if you receive a `model.AppError` you should return it directly (you can see an example of that in the example PR).